### PR TITLE
Using split instead of regex for simple key=value type tags.

### PR DESCRIPTION
**Pull request by c2tarun**
*Issue #, if available:*
Issue #1960 and #1845 

*Why is this change necessary?*
This change will unblock lot of users who use long tag value to upgrade from 0.41.0.

*How does it address the issue?*
For simple `key=value` type values we are not splitting on `=` instead of using regex.

*What side effects does this change have?*
None

*Did you change a dependency in `requirements/base.txt`?*
    *If so, did you run `make update-reproducible-reqs`*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [x] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/cli/types.py [hunk 1]
```diff
@@ -221,3 +226,23 @@ def convert(self, value, param, ctx):
                 )
 
         return result
+
+    @staticmethod
+    def _standard_key_value_parser(tag_value):
+        """
+        Method to parse simple `Key=Value` type tags without using regex. This is similar to how CFN does this.
```
Comment (sriram-mv) [inline samcli/cli/types.py:233]: aws cli, not cfn.

File: samcli/cli/types.py [hunk 2]
```diff
@@ -204,14 +204,19 @@ def convert(self, value, param, ctx):
...
-            groups = re.findall(self._pattern, val)
-
-            if not groups:
-                fail = True
-            for group in groups:
-                key, v = group
-                # assign to result['KeyName1'] = string and so on.
-                result[_unquote_wrapped_quotes(key)] = _unquote_wrapped_quotes(v)
+            # Using standard parser first. We should implement other type parser like JSON and Key=key,Value=val type format.
+            parsed, k, v = self._standard_key_value_parser(val)
```
Comment (sriram-mv) [inline samcli/cli/types.py:208]: should we check for more than 1 count, given that aws cli already does it.

File: samcli/cli/types.py (hunk 1 above)
Comment (c2tarun) [reply samcli/cli/types.py:233]: ahhhh.. missed it. Will fix now.

File: samcli/cli/types.py (hunk 2 above)
Comment (c2tarun) [reply samcli/cli/types.py:208]: aws-cli simply pick first `=`. I think our existing solution already handles more edge cases than aws-cli, we should keep using it until we implement separate parses.

Comment (sriram-mv) [reply samcli/cli/types.py:208]: right, but this solution only looks for one key value pair right?

Comment (c2tarun) [reply samcli/cli/types.py:208]: Nope, we get list of `key=value` tags. We parse each one. They are already separated by command and handled by click.

**Comment by piersf** [comment]
In which release will be this fix available?

**Comment by c2tarun** [comment]
@piersf Code is merged, we haven't finalized exact date yet, but there are few more minor fixes we want to include as well. You can expect this release by early next week.

**Comment by piersf** [comment]
- @c2tarun Thank you! I look forward to the next release.

**Pull request merged**